Tests/StructuredQueriesTests/JSONFunctionsTests.swift

```diff
@@ -339,6 +339,158 @@ extension SnapshotTests {
         """
       }
     }
+
+    @Test func jsonObject() {
+      assertQuery(
+        Reminder
+          .join(RemindersList.all) { $0.remindersListID.eq($1.id) }
+          .select { ($0, $1.jsonObject) }
+      ) {
+        """
+        SELECT "reminders"."id", "reminders"."assignedUserID", "reminders"."dueDate", "reminders"."isCompleted", "reminders"."isFlagged", "reminders"."notes", "reminders"."priority", "reminders"."remindersListID", "reminders"."title", "reminders"."updatedAt", json_object('id', json_quote("remindersLists"."id"), 'color', json_quote("remindersLists"."color"), 'title', json_quote("remindersLists"."title"), 'position', json_quote("remindersLists"."position"))
+        FROM "reminders"
+        JOIN "remindersLists" ON ("reminders"."remindersListID" = "remindersLists"."id")
+        """
+      } results: {
+        #"""
+        ┌─────────────────────────────────────────────┬───────────────────┐
+        │ Reminder(                                   │ RemindersList(    │
+        │   id: 1,                                    │   id: 0,          │
+        │   assignedUserID: 1,                        │   color: 4889071, │
+        │   dueDate: Date(2001-01-01T00:00:00.000Z),  │   title: "",      │
+        │   isCompleted: false,                       │   position: 0     │
```

Comment:
I don't understand why the decoding is behaving strangely here yet. When I manually run a similar query in SQL it prints out the expected JSON.